src/code/z_demo.c

```diff
@@ -1,99 +1,1634 @@
+            }
+            break;
+        case 0x23:
+            if (csCtx->frames != D_801BB15C) {
+                D_801BB15C = csCtx->frames;
+
+                if (REG(15) != 0) {
+                    time = gSaveContext.time;
+                    gSaveContext.time = (u16)REG(15) + time;
+                    time = gSaveContext.time;
+                    gSaveContext.time = (u16)gSaveContext.unk_14 + time;
+                }
+            }
+            break;
+        case 0x24:
+            func_8019F128(NA_SE_EV_EARTHQUAKE_LAST - SFX_FLAG);
+            if (isStartFrame) {
+                sCutsceneQuakeIndex = Quake_Add(GET_ACTIVE_CAM(globalCtx), 6);
+                Quake_SetSpeed(sCutsceneQuakeIndex, 22000);
+                Quake_SetQuakeValues(sCutsceneQuakeIndex, 2, 1, 0, 0);
+                Quake_SetCountdown(sCutsceneQuakeIndex, 800);
+            }
+            break;
+
+        case 0x26:
+            gSaveContext.day = 9;
+
+            {
+                GameState* gameState = &globalCtx->state;
+                gameState->running = false;
+            }
+            SET_NEXT_GAMESTATE(&globalCtx->state, Daytelop_Init, DaytelopContext);
+
+            func_80146F5C(globalCtx);
+            break;
+
+        case 0x27:
+            gSaveContext.playerForm = PLAYER_FORM_ZORA;
+            break;
+
+        case 0x28:
+            csCtx->frames = cmd->startFrame - 1;
+            break;
+    }
+}
+
+// Command 0x97: Set Environment Lighting
+void Cutscene_Command_SetLighting(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdEnvLighting* cmd) {
+    if (csCtx->frames == cmd->startFrame) {
+        if (cmd->setting != 0x20) {
+            globalCtx->envCtx.lightSettingOverride = cmd->setting - 1;
+            globalCtx->envCtx.lightBlend = 1.0f;
+        } else {
+            globalCtx->envCtx.lightSettingOverride = 0xFF;
+        }
+    }
+}
+
+// Command 0x12C: Play Background Music or Fanfare
+void Cutscene_Command_PlaySequence(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdMusicChange* cmd) {
+    if (csCtx->frames == cmd->startFrame) {
+        func_801A2C88(cmd->sequence - 1);
+    }
+}
+
+// Command 0x12D: Stop Background Music or Fanfare
+void Cutscene_Command_StopSequence(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdMusicChange* cmd) {
+    if ((csCtx->frames >= cmd->startFrame) && (cmd->endFrame >= csCtx->frames)) {
+        func_801A2D54(cmd->sequence - 1);
+    }
+}
+
+// Command 0x9C: Fade Background Music or Fanfare over duration
+void Cutscene_Command_FadeSequence(GlobalContext* globalCtx, CutsceneContext* csCtx, CsCmdMusicFade* cmd) {
+    if ((csCtx->frames == cmd->startFrame) && (csCtx->frames < cmd->endFrame)) {
+        u8 frameCount = cmd->endFrame - cmd->startFrame;
+
+        if (cmd->type == 2) {
+            Audio_QueueSeqCmd((frameCount << 0x10) | 0x110000FF);
+        } else {
+            Audio_QueueSeqCmd((frameCount << 0x10) | NA_BGM_STOP);
```

Comment:
```suggestion
        u8 fadeTimer = cmd->endFrame - cmd->startFrame;

        if (cmd->type == 2) {
            Audio_QueueSeqCmd((fadeTimer << 0x10) | 0x110000FF);
        } else {
            Audio_QueueSeqCmd((fadeTimer << 0x10) | NA_BGM_STOP);
```
This would be more appropriate to what the variable is doing